maintenance-editor/src/main/java/ca/on/oicr/gsi/shesmu/maintenance/Editor.java

```diff
@@ -0,0 +1,196 @@
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import javax.swing.BorderFactory;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.JToolBar;
+import javax.swing.UIManager;
+import javax.swing.table.AbstractTableModel;
+import javax.swing.table.TableCellRenderer;
+
+public class Editor extends JFrame {
+  private class DateTableModel extends AbstractTableModel {
+    private static final long serialVersionUID = 1L;
+    final List<LocalDateTime[]> rows = new ArrayList<>();
+
+    @Override
+    public Class<?> getColumnClass(int columnIndex) {
+      return LocalDateTime.class;
+    }
+
+    @Override
+    public int getColumnCount() {
+      return 2;
+    }
+
+    @Override
+    public String getColumnName(int column) {
+      return column == 0 ? "Start" : "End";
+    }
+
+    @Override
+    public int getRowCount() {
+      return rows.size();
+    }
+
+    @Override
+    public Object getValueAt(int row, int col) {
+      return rows.get(row)[col];
+    }
+
+    @Override
+    public boolean isCellEditable(int rowIndex, int columnIndex) {
+      return true;
+    }
+
+    @Override
+    public void setValueAt(Object aValue, int rowIndex, int columnIndex) {
+      rows.get(rowIndex)[columnIndex] = (LocalDateTime) aValue;
+    }
+  }
+
+  private static final long serialVersionUID = 1L;
+  private static final Pattern TAB = Pattern.compile("\t");
+
+  public static void main(String[] args) throws IOException {
+    File file;
+    if (args.length == 0) {
+      final JFileChooser chooser = new JFileChooser();
+      if (chooser.showOpenDialog(null) == JFileChooser.APPROVE_OPTION) {
+        file = chooser.getSelectedFile();
+      } else {
+        return;
+      }
+    } else if (args.length == 1) {
+      file = new File(args[0]);
+    } else {
+      System.err.println("Usage: maintenance-editor [maintenance.tsv]");
+      System.exit(1);
+      return;
+    }
+    new Editor(file).setVisible(true);
+  }
+
+  public Editor(File file) throws IOException {
+    setTitle(file.toString() + " - Maintenance Schedule Editor");
+    setSize(700, 400);
+    setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+    final JPanel panel = new JPanel(new BorderLayout());
+    add(panel);
+    final JToolBar toolbar = new JToolBar();
+    panel.add(toolbar, BorderLayout.PAGE_START);
+    final JButton add = new JButton("Add");
+    toolbar.add(add);
+    final JButton delete = new JButton("Delete");
+    toolbar.add(delete);
+    final JButton reload = new JButton("Revert");
+    toolbar.add(reload);
+    final JButton save = new JButton(UIManager.getIcon("FileView.floppyDriveIcon"));
```

Comment:
The icon is cute but I'm not sure why you need it? You could also use `+` `-` `<-` for the other buttons with appropriate mouseovers.